sim/warlock/soul_fire.go

```diff
@@ -56,6 +58,10 @@ func (warlock *Warlock) registerSoulFire() {
 			baseDamage := warlock.CalcAndRollDamageRange(sim, 2.54299998283, 0.22499999404)
 			result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
 
+			if hasT134P && warlock.SoulBurnAura.IsActive() {
+				warlock.SoulShards += 1
```

Comment:
Needs to be `min(3, warlock.Soulshards + 1)` since you can't have more then 3 shards at once.